linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5986,21 +5957,15 @@ private[optimizer] object OptimizerCore {
       val labelInfos: Map[LabelName, LabelInfo]) {
 
     def withThisLocalDef(rep: LocalDef): OptEnv =
-      withThisLocalDef(Some(rep))
+      withLocalDef(LocalName.This, rep)
 
     def withThisLocalDef(rep: Option[LocalDef]): OptEnv =
-      new OptEnv(rep, localDefs, labelInfos)
+      if (rep.isEmpty) this
```

Comment:
I don't think this branch is correct. IIUC it should remove `LocalName.This` from `localDefs` if it is present.

Reply:
Arguable. It would definitely match what was there before, but IMO either behavior is fine, so I chose the simplest. We never actually use `withThisLocalDef` if `this` already has a binding for `this`. In fact, we otherwise never *remove* bindings from an `OptEnv`. So I would argue that we should preserve the simplest implementation.

Reply:
OK, fair enough. Let's leave it.